File: frontend/src/api/history.ts
import { apiJson } from "./http";

export type HistoryPoint = {
  day: string;
  value: string;
  cash: string;
  securities: string;
  invested: string;
  imoex: string | null;
};

export type HistorySnapshot = {
  building: boolean;
  points: HistoryPoint[];
  xirr_percent: string | null;
  xirr_from: string | null;
};

export function getHistory(refresh = false) {
  const query = refresh ? "?refresh=true" : "";
  return apiJson<HistorySnapshot>(`/api/history${query}`);
}

File: backend/app/schemas/history.py
from datetime import date

from pydantic import BaseModel


class HistoryPointOut(BaseModel):
    day: date
    value: str
    cash: str
    securities: str
    invested: str
    imoex: str | None = None


class HistoryOut(BaseModel):
    building: bool = False
    points: list[HistoryPointOut]
    xirr_percent: str | None = None
    xirr_from: date | None = None
